omnibot/routes/api.py

```diff
@@ -641,31 +641,34 @@ def _perform_action(bot, data):
     )
     parse_kwargs(kwargs, bot)
 
-    if action == 'files.upload':
-        # save base64 encoded file
-        base64_file = kwargs.get('file')
-        base64_bytes = base64_file.encode('ascii')
-        filename = '{}-{}'.format(round(time.time()), kwargs.get('filename'))
-        file = open(filename, 'wb')
-        file.write(base64.decodebytes(base64_bytes))
-        file.close()
-
-        # Open the file for slack to read
-        file = open(filename, 'rb')
-        kwargs['file'] = file
-        ret = slack.client(bot, client_type='oauth_bot').files_upload(**kwargs)
-        file.close()
-
-        # Remove the file
-        os.remove(file.name)
-    else:
-        ret = slack.client(bot, client_type='oauth_bot').api_call(
-            action,
-            json=kwargs
-        )
-    logger.debug(ret)
-    if not ret['ok']:
-        if ret.get('error') in ['missing_scope', 'not_allowed_token_type']:
+    try:
+        if action == 'files.upload':
+            # save base64 encoded file
+            base64_file = kwargs.get('file')
+            base64_bytes = base64_file.encode('ascii')
+            filename = '{}-{}'.format(round(time.time()), kwargs.get('filename'))
+            file = open(filename, 'wb')
+            file.write(base64.decodebytes(base64_bytes))
+            file.close()
+
+            # Open the file for slack to read
+            file = open(filename, 'rb')
+            kwargs['file'] = file
+            ret = slack.client(bot, client_type='oauth_bot').files_upload(**kwargs)
+            file.close()
+
+            # Remove the file
+            os.remove(file.name)
+        else:
+            ret = slack.client(bot, client_type='oauth_bot').api_call(
+                action,
+                json=kwargs
+            )
+        logger.debug(ret)
+    except Exception as e:
+        logger.exception('Error uploading file', kv={'error': str(e)})
+        # TODO (shekharkhedekar) validate errors
+        if str(e) in ['missing_scope', 'not_allowed_token_type']:
```

Comment:
same as above, but you want `e.response.get('error')`

Reply:
updated